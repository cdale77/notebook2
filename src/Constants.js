const Constants = {
  DEFAULT_STATE: {
    session: {
      signedIn: false,
      uuid: ""
    },
    flash: {
      flashType: "",
      message: ""
    }
  },
  ACTIONS: {
    REQUEST_START: "REQUEST_START",
    REQUEST_END: "REQUEST_END",
    FLASH_SUCCESS: "FLASH_SUCCESS",
    FLASH_ERROR: "FLASH_ERROR",
    FLASH_CLEAR: "FLASH_CLEAR",
    NEW_SESSION_SUCCESS: "NEW_SESSION_SUCCESS",
    NEW_SESSION_FAILURE: "NEW_SESSION_FAILURE",
    DESTROY_SESSION: "DESTROY_SESSION"
  }
};

export default Constants;
