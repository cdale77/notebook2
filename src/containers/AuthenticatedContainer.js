import React from "react";
import { connect } from "react-redux";
import Nav from "../components/Nav";
import SessionThunks from "../thunks/SessionThunks";

const mapStateToProps = state => {
  return {
    session: state.session
  };
};

const mapDispatchToProps = dispatch => {
  return {
    signOut: () => {
      dispatch(SessionThunks.signOut());
    }
  };
};

class AuthenticatedContainer extends React.Component {
  render() {
    const signedIn = this.props.session["signedIn"];
    return (
      <div className="authenticated-container">
        <Nav signedIn={signedIn} signOut={this.props.signOut} />
      </div>
    );
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(
  AuthenticatedContainer
);
