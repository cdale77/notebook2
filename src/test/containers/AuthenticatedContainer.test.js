import React from "react";
import Renderer from "react-test-renderer";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import AuthenticatedContainer from "../../containers/AuthenticatedContainer";
import Constants from "../../Constants";
import thunk from "redux-thunk";

const middlewares = [thunk]; // add your middlewares like `redux-thunk`
let state = Constants.DEFAULT_STATE;

describe("AuthenticatedContainer", () => {
  describe("rendering", () => {
    it("should render without error", () => {
      const store = configureMockStore(middlewares)(state);
      const tree = Renderer.create(
        <Provider store={store}>
          <AuthenticatedContainer />
        </Provider>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe("without a currrentBook set", () => {
    it("should render the BookList", () => {
      state["books"] = {
        currentBook: {},
        bookList: [{ bookId: "1234", name: "Foo" }]
      };
      const store = configureMockStore(middlewares)(state);
      const tree = Renderer.create(
        <Provider store={store}>
          <AuthenticatedContainer />
        </Provider>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe("with a currrentBook set", () => {
    it("should render the BookView", () => {
      const book = { bookId: "1234", name: "Foo" };
      state["books"] = {
        currentBook: book,
        bookList: [book]
      };
      const store = configureMockStore(middlewares)(state);

      const tree = Renderer.create(
        <Provider store={store}>
          <AuthenticatedContainer />
        </Provider>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
