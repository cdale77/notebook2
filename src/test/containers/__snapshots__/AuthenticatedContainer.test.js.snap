// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthenticatedContainer rendering should render without error 1`] = `
<div
  className="authenticated-container"
>
  <nav
    className="nav"
  >
    <div
      className="logo"
    >
      NoteBook2
    </div>
  </nav>
  <div
    className="book-list"
  >
    <h2>
      NoteBooks
    </h2>
    <div
      className="form fieldset new-book-form"
    >
      <form>
        <div
          className="form-group"
        >
          <input
            className="input-name"
            id="name"
            name="name"
            onChange={[Function]}
            placeholder="Name"
            type="text"
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="button"
            disabled={false}
            id="submit"
            name="submit"
            onClick={[Function]}
            type="submit"
            value="Submit"
          />
        </div>
      </form>
    </div>
    <div
      className="book-cards"
    />
  </div>
</div>
`;

exports[`AuthenticatedContainer with a currrentBook set should render the BookView 1`] = `
<div
  className="authenticated-container"
>
  <nav
    className="nav"
  >
    <div
      className="logo"
    >
      NoteBook2
    </div>
  </nav>
  <div
    className="book-view"
  >
    <h2>
      Foo
    </h2>
    <div
      className="book-view"
    >
      BookView
    </div>
  </div>
</div>
`;

exports[`AuthenticatedContainer without a currrentBook set should render the BookList 1`] = `
<div
  className="authenticated-container"
>
  <nav
    className="nav"
  >
    <div
      className="logo"
    >
      NoteBook2
    </div>
  </nav>
  <div
    className="book-list"
  >
    <h2>
      NoteBooks
    </h2>
    <div
      className="form fieldset new-book-form"
    >
      <form>
        <div
          className="form-group"
        >
          <input
            className="input-name"
            id="name"
            name="name"
            onChange={[Function]}
            placeholder="Name"
            type="text"
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="button"
            disabled={false}
            id="submit"
            name="submit"
            onClick={[Function]}
            type="submit"
            value="Submit"
          />
        </div>
      </form>
    </div>
    <div
      className="book-cards"
    >
      <div
        className="book-listing"
        onClick={[Function]}
      >
        <div
          className="book-name"
        >
          Foo
        </div>
      </div>
    </div>
  </div>
</div>
`;
